spec/e2e/pooling.spec.ts

```diff
@@ -0,0 +1,63 @@
+import { IrcBridgeE2ETest } from "../util/e2e-test";
+import { IrcConnectionPool } from '../../src/pool-service/IrcConnectionPool';
+
+const redisUrl = process.env.IRCBRIDGE_TEST_REDIS_URL ?? 'redis://localhost:6379';
+
+describe('Connection pooling', () => {
+    let server: IrcBridgeE2ETest;
+    let pool: IrcConnectionPool;
+    beforeEach(async () => {
+        pool = new IrcConnectionPool({
+            redisUri: redisUrl,
+            metricsHost: false,
+            metricsPort: 7002,
+            loggingLevel: 'debug',
+        });
+        server = await IrcBridgeE2ETest.createTestEnv({
+            matrixLocalparts: ['alice'],
+            ircNicks: ['bob'],
+            config: {
+                connectionPool: {
+                    redisUrl,
+                }
+            }
+        });
+        pool.main();
+        await server.setUp();
+    });
+
+    afterEach(async () => {
+        await Promise.allSettled([
+            server.tearDown(),
+            pool.close(),
+        ]);
+    })
+
+    it('should be able to dynamically bridge a room via the !join command', async () => {
```

Comment:
Rather than copying the test, run all the tests twice: One with single process, one as a multi-process.